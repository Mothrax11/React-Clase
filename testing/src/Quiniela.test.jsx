import { render, screen, fireEvent } from "@testing-library/react";
import Quiniela from "./Quiniela";

test("debería renderizar Quiniela correctamente", () => {
    const { asFragment } = render(<Quiniela />);
    expect(asFragment()).toMatchSnapshot();
});

test("debería guardar la nueva apuesta", () => {
    render(<Quiniela />);
    const button = screen.getByText(/Nueva Apuesta/i);
    fireEvent.click(button);

    fireEvent.change(screen.getByLabelText(/Fecha/i), { target: { value: "2025-03-10" } });
    fireEvent.change(screen.getByLabelText(/Jornada/i), { target: { value: "1" } });
    fireEvent.change(screen.getByLabelText(/Equipo Local/i), { target: { value: "Equipo A" } });
    fireEvent.change(screen.getByLabelText(/Equipo Visitante/i), { target: { value: "Equipo B" } });
    fireEvent.change(screen.getByLabelText(/Apuesta 1/i), { target: { value: "1" } });

    fireEvent.click(screen.getByText(/Guardar/i));

    expect(screen.getByText("Equipo A VS Equipo B")).toBeInTheDocument();
});
