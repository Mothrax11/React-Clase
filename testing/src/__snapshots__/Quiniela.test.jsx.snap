// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deber√≠a renderizar Quiniela correctamente 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="btn btn-primary"
      type="button"
    >
      Nueva Apuesta
    </button>
    <div />
  </div>
</DocumentFragment>
`;
